services-toolkit/how-to-guides/configure-dynamic-provisioning-rds.hbs.md

```diff
@@ -0,0 +1,208 @@
+   name: rdsinstance
+   patches:
+   - fromFieldPath: metadata.uid
+     toFieldPath: spec.writeConnectionSecretToRef.name
+     transforms:
+     - string:
+         fmt: '%s-postgresql'
+         type: Format
+       type: string
+     type: FromCompositeFieldPath
+   - fromFieldPath: spec.storageGB
+     toFieldPath: spec.forProvider.allocatedStorage
+     type: FromCompositeFieldPath
+```
+
+This `Composition` is intended to be used as a starting point only. You will likely need to update the `forProvider` configuration to your specific requirements. In particular note the `publiclyAccessible: true` setting, which has the effect of opening up public access to all dynamically provisioned RDS databases. You will most likely want to disable this setting and to instead configure internal connectivity. You may wish to seek inspiration from [this example](https://marketplace.upbound.io/configurations/xp/getting-started-with-aws-with-vpc/latest/compositions/vpcpostgresqlinstances.aws.database.example.org/database.example.org/XPostgreSQLInstance), which defines a composition that creates a separate VPC for each RDS PostgreSQL instance and automatically configures inbound rules.
+
+Once you are happy with the configuration, use `kubectl` to apply the file to the TAP cluster.
+
+```console
+$ kubectl apply -f xpostgresqlinstances.database.rds.example.org.composition.yml
+```
+
+## <a id="stk-dynamic-provisioning-rds-step-4"></a> Step 4: Make the service discoverable to application teams
+
+Create a file named `rds.class.yml` and copy in the following contents.
+
+```yaml
+# rds.class.yml
+
+---
+apiVersion: services.apps.tanzu.vmware.com/v1alpha1
+kind: ClusterInstanceClass
+metadata:
+  name: aws-rds-psql
+spec:
+  description:
+    short: Amazon AWS RDS PostgreSQL
+  provisioner:
+    crossplane:
+      compositeResourceDefinition: xpostgresqlinstances.database.rds.example.org
+```
+
+Then use `kubectl` to apply the file to the TAP cluster.
+
+```console
+$ kubectl apply -f rds.class.yml
+```
+
+## <a id="stk-dynamic-provisioning-rds-step-5"></a> Step 5: Authorize users with the app-operator role to claim from the class
+
+Create a file named `app-operator-claim-aws-rds-psql.rbac.yml` and copy in the following contents.
+
+```yaml
+# app-operator-claim-aws-rds-psql.rbac.yml
+
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: app-operator-claim-aws-rds-psql
+  labels:
+    apps.tanzu.vmware.com/aggregate-to-app-operator-cluster-access: "true"
+rules:
+- apiGroups:
+  - "services.apps.tanzu.vmware.com"
+  resources:
+  - clusterinstanceclasses
+  resourceNames:
+  - aws-rds-psql
+  verbs:
+  - claim
+```
+
+Then use `kubectl` to apply the file to the TAP cluster.
+
+```console
+$ kubectl apply -f app-operator-claim-aws-rds-psql.rbac.yml
+```
+
+## <a id="stk-dynamic-provisioning-rds-step-6"></a> Step 6: Take it for a spin
+
+```console
+$ tanzu service class-claim create rds-psql-1 --class aws-rds-psql -p storageGB=30
```

Comment:
**suggestion** This RDS instance won't be connectable from the internet, someone has to create a `securityGroup` rule to allow inbound traffic. To start with perhaps we could have a note to that effect. In a 2nd pass we can include that in the composition.

Reply:
Yeah I tried to cover that conundrum with the text earlier in the guide which says something like "you might want to setup internal connectivity following this other example from Upbound"... But perhaps a note at this step would be useful as well. I'll add that now.